source/cl/test/UnitCL/include/cl_intel_unified_shared_memory.h

```diff
@@ -51,11 +60,116 @@ struct cl_intel_unified_shared_memory_Test : public virtual ucl::ContextTest {
 #undef GET_EXTENSION_ADDRESS
   }
 
+  void TearDown() override {
+    if (device_ptr) {
+      cl_int err = clMemBlockingFreeINTEL(context, device_ptr);
+      EXPECT_SUCCESS(err);
+      device_ptr = nullptr;
+    }
+
+    if (shared_ptr) {
+      cl_int err = clMemBlockingFreeINTEL(context, shared_ptr);
+      EXPECT_SUCCESS(err);
+      shared_ptr = nullptr;
+    }
+
+    if (host_ptr) {
+      cl_int err = clMemBlockingFreeINTEL(context, host_ptr);
+      EXPECT_SUCCESS(err);
+      host_ptr = nullptr;
+    }
```

Comment:
What does `EXPECT_SUCCESS` do on error? If it aborts the process, or if it prints a warning and carries on, you can use your `allPointers()` helper function here.
```suggestion
    for (auto *ptr : allPointers()) {
      cl_int err = clMemBlockingFreeINTEL(context, device_ptr);
      EXPECT_SUCCESS(err);
    }
    device_ptr = shared_ptr = host_ptr = nullptr;
```

Reply:
`EXPECT_*` functions continue the test while `ASSERT_*` functions halt it instantly (but only if the calling function has a specific return type). The helper function would work here, thanks.